runtime/autoload/spellfile.lua

```diff
@@ -0,0 +1,137 @@
+local M = {}
+
+M.config = {
+  url = 'https://ftp.nluug.nl/pub/vim/runtime/spell',
+  encoding = vim.o.encoding,
+  rtp = vim.opt.rtp:get(),
+}
+
+M.done = {}
+
+M.setup = function(opts)
+  M.done = {}
+  M.config = vim.tbl_extend('force', M.config, opts or {})
+end
+
+M.directory_choices = function()
+  local options = {}
+  for _, dir in ipairs(M.config.rtp) do
+    local spell = dir .. '/spell'
+    if vim.fn.isdirectory(spell) == 1 then
+      table.insert(options, spell)
+    end
+  end
+  return options
+end
+
+M.choose_directory = function()
+  local options = M.directory_choices()
+  if #options == 0 then
+    vim.notify('No spell directory found in the runtimepath')
+    return
+  elseif #options == 1 then
+    return options[1]
+  end
+
+  local prompt = {}
+  for idx, dir in pairs(options) do
+    table.insert(prompt, string.format('%d: %s', idx, dir))
+  end
+  local choice = vim.fn.inputlist(prompt)
+  if choice <= 0 or choice > #options then
+    return
+  end
+
+  return options[choice]
+end
+
+M.exists = function(file_name)
+  local is_file = function(pth)
+    local stat = vim.loop.fs_stat(pth)
+    return stat ~= nil and stat.type ~= nil and stat.type == 'file'
+  end
+
+  for _, dir in pairs(M.directory_choices()) do
```

Comment:
prefer `vim.fs` functions for iterating

Reply:
I'm gonna study `vim.fs` to see how to refactor. Probably gonna wait until I have tests working in this branch.